pkg/controller/cronjob/utils.go

```diff
@@ -321,3 +321,16 @@ func adoptJobs(sj *batchv2alpha1.CronJob, js []batchv1.Job, jc jobControlInterfa
 	}
 	return utilerrors.NewAggregate(errs)
 }
+
+func getCurrentTimeInZone(sj *batchv2alpha1.CronJob) time.Time {
+	if sj.Spec.Timezone == nil {
+		return time.Now()
+	}
+
+	loc, err := time.LoadLocation(*sj.Spec.Timezone)
+	if err != nil {
+		loc = time.Local
```

Comment:
So LoadLocation can fail only because the timezone specified is invalid? If so, that should be a validation error probably or we shouldn't run the job at all.